### ISSUE TITLE
Switching branches in git causes multiple builds

### ISSUE THREAD

**Issue by zachleat** [11ty/eleventy#3077]
Not sure if this is a docs issue or a code issue (I’d prefer to fix in code!)

With `--incremental`:

<img width="707" alt="image" src="https://github.com/11ty/eleventy/assets/39355/a4521af2-a005-45de-b156-6b246ea0e0e7">

Without `--incremental`:

<img width="565" alt="image" src="https://github.com/11ty/eleventy/assets/39355/e7c94ee6-f09d-4c81-925f-08d979c54130">

Usually we’ll recommend something like `eleventyConfig.setWatchThrottleWaitTime(500);` https://www.11ty.dev/docs/watch-serve/#add-delay-before-re-running but that doesn’t seem to be working here either.

Desired behavior: one build!

**Comment by zachleat** [comment]
Logs with eleventy-base-blog on Eleventy v2.0.1:

```
> eleventy-base-blog@8.0.0 start
> npx @11ty/eleventy --serve --quiet --incremental

[11ty/eleventy-base-blog] Including drafts.
[11ty] Copied 5 files / Wrote 17 files in 0.18 seconds (10.6ms each, v2.0.1)
[11ty] Watching…
[11ty] Server at http://localhost:8080/
[11ty] File changed: content/blog/secondpost.md
[11ty] File changed: content/blog/fourthpost/fourthpost.md
[11ty] File changed: public/css/index.css
[11ty] File changed: content/blog/firstpost.md
[11ty] File changed: content/blog/thirdpost.md
[11ty] Benchmark      4ms  22%    13× (Configuration) "slugify" Nunjucks Filter
[11ty] Wrote 1 file (skipped 13) in 0.02 seconds (v2.0.1)
[11ty] Benchmark      2ms  17%    13× (Configuration) "slugify" Nunjucks Filter
[11ty] Wrote 1 file (skipped 13) in 0.01 seconds (v2.0.1)
[11ty] Copied 1 file / Wrote 0 files (skipped 14) in 0.00 seconds (v2.0.1)
[11ty] Benchmark      4ms  27%    13× (Configuration) "slugify" Nunjucks Filter
[11ty] Wrote 1 file (skipped 13) in 0.02 seconds (v2.0.1)
[11ty] Benchmark      2ms  18%    14× (Configuration) "slugify" Nunjucks Filter
[11ty] Wrote 1 file (skipped 13) in 0.01 seconds (v2.0.1)
```

```
> eleventy-base-blog@8.0.0 start
> npx @11ty/eleventy --serve --quiet

[11ty/eleventy-base-blog] Including drafts.
[11ty] Copied 5 files / Wrote 17 files in 0.18 seconds (10.6ms each, v2.0.1)
[11ty] Watching…
[11ty] Server at http://localhost:8080/
[11ty] File changed: content/blog/thirdpost.md
[11ty] File changed: content/blog/fourthpost/fourthpost.md
[11ty] File changed: content/blog/firstpost.md
[11ty] File changed: content/blog/secondpost.md
[11ty] File changed: public/css/index.css
[11ty] Benchmark      5ms  14%    21× (Configuration) "slugify" Nunjucks Filter
[11ty] Copied 5 files / Wrote 17 files in 0.04 seconds (2.4ms each, v2.0.1)
[11ty] File changed: content/blog/fourthpost/fourthpost.md
[11ty] File changed: content/blog/secondpost.md
[11ty] File changed: content/blog/firstpost.md
[11ty] File changed: content/blog/thirdpost.md
[11ty] File changed: public/css/index.css
[11ty] Benchmark      6ms  12%    21× (Configuration) "slugify" Nunjucks Filter
[11ty] Copied 5 files / Wrote 17 files in 0.05 seconds (2.9ms each, v2.0.1)
[11ty] Benchmark     16ms  23%    21× (Configuration) "slugify" Nunjucks Filter
[11ty] Copied 5 files / Wrote 17 files in 0.07 seconds (4.1ms each, v2.0.1)
```

**Comment by zachleat** [comment]
Logs with eleventy-base-blog on v3.0.0-alpha.13:

```
> eleventy-base-blog@9.0.0 start
> npx @11ty/eleventy --serve --quiet --incremental

[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Copied 6 files / Wrote 16 files in 0.18 seconds (11.3ms each, v3.0.0-alpha.13)
[11ty] Watching…
[11ty] Server at http://localhost:8080/
[11ty] File changed: content/blog/firstpost.md
[11ty] File changed: content/blog/fourthpost/fourthpost.md
[11ty] File changed: content/blog/thirdpost.md
[11ty] File changed: content/blog/secondpost.md
[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Benchmark      8ms  22%     9× (Configuration) "slugify" JavaScript Function
[11ty] Wrote 2 files (skipped 11) in 0.04 seconds (v3.0.0-alpha.13)
[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Benchmark      7ms  29%     9× (Configuration) "slugify" JavaScript Function
[11ty] Wrote 2 files (skipped 11) in 0.03 seconds (v3.0.0-alpha.13)
[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Benchmark      7ms  32%    12× (Configuration) "slugify" JavaScript Function
[11ty] Wrote 2 files (skipped 11) in 0.02 seconds (v3.0.0-alpha.13)
[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Benchmark      5ms  28%     9× (Configuration) "slugify" JavaScript Function
[11ty] Wrote 2 files (skipped 11) in 0.02 seconds (v3.0.0-alpha.13)
```

```
> eleventy-base-blog@9.0.0 start
> npx @11ty/eleventy --serve --quiet

[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Copied 6 files / Wrote 16 files in 0.23 seconds (14.4ms each, v3.0.0-alpha.13)
[11ty] Watching…
[11ty] Server at http://localhost:8080/
[11ty] File changed: content/blog/thirdpost.md
[11ty] File changed: content/blog/secondpost.md
[11ty] File changed: content/blog/firstpost.md
[11ty] File changed: content/blog/fourthpost/fourthpost.md
[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Benchmark     19ms  20%    17× (Configuration) "slugify" JavaScript Function
[11ty] Copied 6 files / Wrote 16 files in 0.09 seconds (5.6ms each, v3.0.0-alpha.13)
[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Benchmark     12ms  22%    17× (Configuration) "slugify" JavaScript Function
[11ty] Benchmark      6ms  12%     9× (Configuration) "slugify" Nunjucks Filter
[11ty] Copied 6 files / Wrote 16 files in 0.05 seconds (3.1ms each, v3.0.0-alpha.13)
```

**Comment by zachleat** [comment]
This is now fixed for non-incremental builds via 65de3af07c25418a984ee77178736c86d920d7e3:

```
[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Copied 6 files / Wrote 16 files in 0.18 seconds (11.3ms each, v3.0.0-alpha.14)
[11ty] Watching…
[11ty] Server at http://localhost:8080/
[11ty] File changed: content/blog.njk
[11ty] File changed: content/blog/secondpost.md
[11ty] File changed: content/blog/firstpost.md
[11ty] File changed: content/blog/fourthpost/fourthpost.md
[11ty] File changed: content/blog/thirdpost.md
[11ty] File changed: content/index.njk
[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Copied 6 files / Wrote 16 files in 0.10 seconds (6.3ms each, v3.0.0-alpha.14)
[11ty] File changed: content/blog.njk
[11ty] File changed: content/blog/secondpost.md
[11ty] File changed: content/blog/firstpost.md
[11ty] File changed: content/blog/thirdpost.md
[11ty] File changed: content/index.njk
[11ty] File changed: content/blog/fourthpost/fourthpost.md
[11ty/eleventy-img] 1 image optimized (1 deferred)
[11ty] Copied 6 files / Wrote 16 files in 0.08 seconds (5.0ms each, v3.0.0-alpha.14)
```

The larger issue here is that `--incremental` builds only support processing one template at a time in the queue. I think we can make that better, but I filed that as a new issue #3334 outside of the 3.0 milestone.

**Comment by zachleat** [comment]
Fixed in v4.0.0-alpha.8 for incremental builds, track at #4274
